<a href="https://colab.research.google.com/github/Shivamgond2022/demo/blob/main/Assignment_3(Q4).ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain %  of the customer order forms. Any error in order form renders it defective and has to be reworked before processing.  The manager wants to check whether the defective %  varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences



In [ ]:
import pandas as pd
import numpy as np
from scipy import stats

In [ ]:
data = pd.read_csv("/content/Costomer+OrderForm.csv")
data

Phillippines   Indonesia       Malta       India
0     Error Free  Error Free   Defective  Error Free
1     Error Free  Error Free  Error Free   Defective
2     Error Free   Defective   Defective  Error Free
3     Error Free  Error Free  Error Free  Error Free
4     Error Free  Error Free   Defective  Error Free
..           ...         ...         ...         ...
295   Error Free  Error Free  Error Free  Error Free
296   Error Free  Error Free  Error Free  Error Free
297   Error Free  Error Free   Defective  Error Free
298   Error Free  Error Free  Error Free  Error Free
299   Error Free   Defective   Defective  Error Free

[300 rows x 4 columns]

In [ ]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 300 entries, 0 to 299
Data columns (total 4 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   Phillippines  300 non-null    object
 1   Indonesia     300 non-null    object
 2   Malta         300 non-null    object
 3   India         300 non-null    object
dtypes: object(4)
memory usage: 9.5+ KB


In [ ]:
data.value_counts()

Phillippines  Indonesia   Malta       India     
Error Free    Error Free  Error Free  Error Free    206
Defective     Error Free  Error Free  Error Free     21
Error Free    Defective   Error Free  Error Free     20
              Error Free  Defective   Error Free     18
                          Error Free  Defective      16
              Defective   Defective   Error Free      9
Defective     Error Free  Defective   Error Free      4
              Defective   Error Free  Error Free      2
              Error Free  Error Free  Defective       2
Error Free    Defective   Error Free  Defective       2
dtype: int64

In [ ]:
# Counting value of all  individual columns
data["Phillippines"].value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [ ]:
data["Indonesia"].value_counts()

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [ ]:
data["Malta"].value_counts()

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [ ]:
data["India"].value_counts()

Error Free    280
Defective      20
Name: India, dtype: int64

In [ ]:
values = np.array([[271,267,269,280],[29,33,31,20]])
values

array([[271, 267, 269, 280],
       [ 29,  33,  31,  20]])

In [ ]:
results = stats.chi2_contingency(values)
results

(3.858960685820355,
 0.2771020991233135,
 3,
 array([[271.75, 271.75, 271.75, 271.75],
        [ 28.25,  28.25,  28.25,  28.25]]))

In [ ]:
results[1]

0.2771020991233135

# Conclusion 
 
 We are fail to reject null hypothesis because our p_value == 0.277 whish is grater than our alpha significant value == 0.05. That menas our all propertis are equal 